packages/sitecore-jss/src/site/graphql-siteinfo-service.ts

```diff
@@ -0,0 +1,132 @@
+      results {
+      ... on Site_e46f3af239fa4866a1577017c4b2a40c {
+        name
+        hostName {
+          value
+        }
+        virtualFolder {
+          value
+        }
+        language: field(name: "Language") {
+          value
+        }
+      }
+    }
+  }
+}
+`;
+
+export type SiteInfo = {
+  name: string;
+  hostName: string;
+  virtualFolder: string;
+  language: string;
+};
+
+export type GraphQLSiteInfoServiceConfig = CacheOptions & {
+  /**
+   * Your Graphql endpoint
+   */
+  endpoint: string;
+  /**
+   * The API key to use for authentication
+   */
+  apiKey: string;
+};
+
+type GraphQLSiteInfoResponse = {
+  search: {
+    results: GraphQLSiteInfoResult[];
+  };
+};
+
+type GraphQLSiteInfoResult = {
+  name: string;
+  hostName: {
+    value: string;
+  };
+  virtualFolder: {
+    value: string;
+  };
+  language: {
+    value: string;
+  };
+};
+
+export class GraphQLSiteInfoService {
+  private graphQLClient: GraphQLClient;
+  private cache: CacheClient<SiteInfo[]>;
+
+  protected get query(): string {
+    return defaultQuery;
+  }
+
+  /**
+   * Creates an instance of graphQL service to retrieve site configuration list from Sitecore
+   * @param {GraphQLSiteInfoServiceConfig} config instance
+   */
+  constructor(private config: GraphQLSiteInfoServiceConfig) {
+    this.graphQLClient = this.getGraphQLClient();
+    this.cache = this.getCacheClient();
+  }
+
+  async fetchSiteInfo(): Promise<SiteInfo[]> {
+    const cachedResult = this.cache.getCacheValue(this.getCacheKey());
+    if (cachedResult) {
+      return cachedResult;
+    }
+
+    const response = await this.graphQLClient.request<GraphQLSiteInfoResponse>(this.query);
+    const result = response?.search?.results?.reduce<SiteInfo[]>((result, current) => {
+      result.push({
+        name: current.name,
+        hostName: current.hostName.value,
+        virtualFolder: current.virtualFolder.value,
+        language: current.language.value,
+      });
+      return result;
+    }, []);
+    this.cache.setCacheValue(this.getCacheKey(), result);
+    return result;
+  }
+
+  /**
+   * Gets cache client implementation
+   * Override this method if custom cache needs to be used
+   * @returns CacheClient instance
+   */
+  protected getCacheClient(): CacheClient<SiteInfo[]> {
+    return new MemoryCacheClient<SiteInfo[]>({
+      cacheEnabled: this.config.cacheEnabled ?? true,
+      cacheTimeout: this.config.cacheTimeout ?? 10,
+    });
+  }
+
+  /**
+   * Gets a GraphQL client that can make requests to the API. Uses graphql-request as the default
+   * library for fetching graphql data (@see GraphQLRequestClient). Override this method if you
+   * want to use something else.
+   * @returns {GraphQLClient} implementation
+   */
+  protected getGraphQLClient(): GraphQLClient {
+    return new GraphQLRequestClient(this.config.endpoint, {
+      apiKey: this.config.apiKey,
+      debugger: debug('multi-site'), // TODO replace with global from debug
```

Comment:
Feel free to add this to debug. Maybe "multisite" to stick with current naming convention of namespaces.